File: PreUtil/src/output/python/easy/ValidPalindromeIi.py
#https://leetcode.com/problems/valid-palindrome-ii/
#Given a string s, return true if the s can be palindrome after deleting at most one character from it.


class Solution(object):
    def validPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """       
        if s==s[::-1]:
            return True
        else:
            for i in range(len(s)):
                if s[i]!=s[len(s)-1-i]:
                    return s[i:len(s)-1-i]==s[i:len(s)-1-i][::-1] or s[i+1:len(s)-i]==s[i+1:len(s)-i][::-1]
        return False
        

    

File: Tool/PreUtil/src/output/javascript/easy/ValidPalindromeIi.js
/**https://leetcode.com/problems/valid-palindrome-ii/ */
//Given a string s, return true if the s can be palindrome after deleting at most one character from it.


/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function(s) {
    let i = 0;
    let j = s.length - 1;
    while(i < j){
        if(s[i] != s[j]){
            return isPalindrome(s, i + 1, j) || isPalindrome(s, i, j - 1);
        }
        i++;
        j--;
    }
    return true;
    
    
};

    